In [1]:
import pandas as pd 
import numpy as np 
from datetime import datetime
from sklearn.linear_model import LinearRegression
from sklearn.model_selection import train_test_split
from sklearn.metrics import r2_score, mean_squared_error

<a id = "summary"></a>
# The Project title 
<ol> 
    <li><a href = "#introduction">Introduction</a></li>
    <li><a href = "#business">Business Understanding</a></li>
    <li><a href = "#data_u">Data Understanding</a></li>
    <li><a href = "#data_p">Data Preparation</a></li>
        <ol>
            <li> 
                <a href = "#intro"> First Part</a>
            </li>
            <li> 
                <a href = "#intro2"> Second Part</a>
            </li>
            <li> 
                <a href = "#third"> Third Part</a>
            </li>
            <li> 
                <a href = "#remain"> Remaining Parts</a>
            </li>
        </ol>
   <li><a href = "#modeling">Modeling</a></li>
   <li><a href = "#evaluation">Evaluation</a></li>
</ol>


<a id = "introduction" ></a>
## Introduction

<a href = "#summary"> Link to summary</a>

In this notebook I am attempting to answer the following questions using the data provided by the Airbnb

#### 1. What are the factors that influence the price of the apartments ? 


#### 2. What are the neighbourhoods with the highest priced appartments ? 


#### 3. What are the neighbourhoods with the lowest priced appartments ? 

All this data is related to Berlin, the city in which I currently work. 
This notebook was prepared as part of the Udacity's  Data Scientist programm. 

In this Notebook I am attempting to follow the CRISP-DM process. 

The first phase of the project is relatively straight forward: the business model of Airbnb is known, also the data and the attached data dictionary are self explanatory. 


For the <b>first question</b> I attempt to use Machine Learning to create model in order to predict the price of a given apartment listing on Airbnb 
If this does not work, I will attempt to see what are the factors that are mostly correlated to Price. 

For <b> the second and third question</b> I simply manipulate the data to extract the means by neighbourhood which are eventualy order in a top10 fashion


<a id = "business"></a>
## Business Understanding
<a href = "#summary"> Link to summary</a>
<br />
When talking about the business understanding, in this notebook I will be analysing the Airbnb data related to Berlin. As it is available at 12th of March 2021. You can find more about airbnb as a company from 
<a href="https://en.wikipedia.org/wiki/Airbnb" >here </a>

<a id = "data_u"></a>
## Data Understanding
<a href = "#summary"> Link to summary</a>

<p> 
Fortunately Airbnb provides us with the data dictionary that is available <a href = "https://docs.google.com/spreadsheets/d/1iWCNJcSutYqpULSQHlNyGInUvHg2BoUGoNRIGa6Szc4/edit#gid=982310896">here</a>
</p>


In [2]:
# importing the data
# importing of the data takes place directly from the Aribnb's website
# url 
url = "http://data.insideairbnb.com/germany/be/berlin/2021-03-12/data/listings.csv.gz"
df_listings = pd.read_csv(url)

In [3]:
df_listings.head()

id                        listing_url       scrape_id last_scraped  \
0  1944  https://www.airbnb.com/rooms/1944  20210312174442   2021-03-14   
1  2015  https://www.airbnb.com/rooms/2015  20210312174442   2021-03-20   
2  3176  https://www.airbnb.com/rooms/3176  20210312174442   2021-03-15   
3  3309  https://www.airbnb.com/rooms/3309  20210312174442   2021-03-14   
4  6883  https://www.airbnb.com/rooms/6883  20210312174442   2021-03-20   

                                                name  \
0         bright & airy Pberg/Mitte 3 months or more   
1   Berlin-Mitte Value! Quiet courtyard/very central   
2                    Fabulous Flat in great Location   
3                  BerlinSpot Schöneberg near KaDeWe   
4  Stylish East Side Loft in Center with AC & 2 b...   

                                         description  \
0  Private, bright and friendly room. You'd be sh...   
1  Great location!  <br />30 of 75 sq meters. Thi...   
2  This beautiful first floor apartment  is situa...   
3  First of all:<br />I prefer short-notice booki...   
4  This apartment is available for people working...   

                               neighborhood_overview  \
0  near all the trendy cafés and flea markets and...   
1  It is located in the former East Berlin area o...   
2  The neighbourhood is famous for its variety of...   
3  My flat is in the middle of West-Berlin, direc...   
4  The emerging and upcoming East of the new hip ...   

                                         picture_url  host_id  \
0  https://a0.muscache.com/pictures/107924758/1f4...     2164   
1  https://a0.muscache.com/pictures/260fd609-710a...     2217   
2  https://a0.muscache.com/pictures/243355/84afcf...     3718   
3  https://a0.muscache.com/pictures/29054294/b1fa...     4108   
4  https://a0.muscache.com/pictures/06b77d9b-4843...    16149   

                                  host_url  ... review_scores_communication  \
0   https://www.airbnb.com/users/show/2164  ...                         9.0   
1   https://www.airbnb.com/users/show/2217  ...                        10.0   
2   https://www.airbnb.com/users/show/3718  ...                         9.0   
3   https://www.airbnb.com/users/show/4108  ...                        10.0   
4  https://www.airbnb.com/users/show/16149  ...                        10.0   

  review_scores_location review_scores_value            license  \
0                    9.0                 8.0                NaN   
1                   10.0                 9.0                NaN   
2                   10.0                 9.0                NaN   
3                    9.0                 9.0                NaN   
4                   10.0                10.0  02/Z/RA/008250-18   

  instant_bookable calculated_host_listings_count  \
0                f                              3   
1                f                              6   
2                f                              1   
3                f                              1   
4                f                              1   

  calculated_host_listings_count_entire_homes  \
0                                           1   
1                                           6   
2                                           1   
3                                           0   
4                                           1   

  calculated_host_listings_count_private_rooms  \
0                                            2   
1                                            0   
2                                            0   
3                                            1   
4                                            0   

  calculated_host_listings_count_shared_rooms reviews_per_month  
0                                           0              0.20  
1                                           0              2.24  
2                                           0              1.03  
3                                           0              0.29  
4                            

In [4]:
# a look at the columns
df_listings.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 19858 entries, 0 to 19857
Data columns (total 74 columns):
 #   Column                                        Non-Null Count  Dtype  
---  ------                                        --------------  -----  
 0   id                                            19858 non-null  int64  
 1   listing_url                                   19858 non-null  object 
 2   scrape_id                                     19858 non-null  int64  
 3   last_scraped                                  19858 non-null  object 
 4   name                                          19826 non-null  object 
 5   description                                   19292 non-null  object 
 6   neighborhood_overview                         10667 non-null  object 
 7   picture_url                                   19858 non-null  object 
 8   host_id                                       19858 non-null  int64  
 9   host_url                                      19858 non-null 

In [5]:
df_listings = df_listings.iloc[:, np.r_[0, 11, 13:21, 22,  25:28, 32:35, 36, 37, 38,39, 40, 41, 55, 56, 60:69, 70]]

#### Also we will be looking only at the entire appartments 
We do this because the focus of this notebook is not the rooms in the apartment, I am interested in the entire apartment. 

In [6]:
df_listings = df_listings[df_listings.room_type == "Entire home/apt"]

<a id = "data_p"></a>
## Data Preparation

<a href = "#summary"> Link to summary</a>

<p>
    By data preparation we mean the eleminating of the null values and formating the data in a way suitable to enable machine learning.
</p>

In [7]:
# a general look at the data
df_listings.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 10927 entries, 1 to 19856
Data columns (total 35 columns):
 #   Column                                       Non-Null Count  Dtype  
---  ------                                       --------------  -----  
 0   id                                           10927 non-null  int64  
 1   host_since                                   10401 non-null  object 
 2   host_about                                   5536 non-null   object 
 3   host_response_time                           5061 non-null   object 
 4   host_response_rate                           5061 non-null   object 
 5   host_acceptance_rate                         6024 non-null   object 
 6   host_is_superhost                            10401 non-null  object 
 7   host_thumbnail_url                           10401 non-null  object 
 8   host_picture_url                             10401 non-null  object 
 9   host_neighbourhood                           7336 non-null   object 
 10

From the above info table we see that there are 10927 rows. A peculiar thing is the fact that the number 10401 gets repeated as non-null values, therefore, the starting point in cleaning the data will be these columns 

In [8]:
# selecting the rows with the 10401 non-null rows
df_listings[df_listings.columns[df_listings.notnull().sum() == 10401]].info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 10927 entries, 1 to 19856
Data columns (total 6 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   host_since                 10401 non-null  object 
 1   host_is_superhost          10401 non-null  object 
 2   host_thumbnail_url         10401 non-null  object 
 3   host_picture_url           10401 non-null  object 
 4   host_total_listings_count  10401 non-null  float64
 5   host_identity_verified     10401 non-null  object 
dtypes: float64(1), object(5)
memory usage: 597.6+ KB


In [9]:
# checking on the first row 
df_listings[df_listings.columns[df_listings.notnull().sum() == 10401]].head(1)

host_since host_is_superhost  \
1  2008-08-18                 f   

                                  host_thumbnail_url  \
1  https://a0.muscache.com/im/pictures/user/21428...   

                                    host_picture_url  \
1  https://a0.muscache.com/im/pictures/user/21428...   

   host_total_listings_count host_identity_verified  
1                        6.0                      t

In [10]:
# assigning to a new df for ease of visualization 
df_10401 = df_listings[df_listings.columns[df_listings.notnull().sum() == 10401]]

In [11]:
# looking at nan values
df_10401[df_10401.host_identity_verified.isna()]

host_since host_is_superhost host_thumbnail_url host_picture_url  \
19           NaN               NaN                NaN              NaN   
53           NaN               NaN                NaN              NaN   
59           NaN               NaN                NaN              NaN   
120          NaN               NaN                NaN              NaN   
143          NaN               NaN                NaN              NaN   
...          ...               ...                ...              ...   
19727        NaN               NaN                NaN              NaN   
19821        NaN               NaN                NaN              NaN   
19824        NaN               NaN                NaN              NaN   
19839        NaN               NaN                NaN              NaN   
19849        NaN               NaN                NaN              NaN   

       host_total_listings_count host_identity_verified  
19                           NaN                    NaN  
53                           NaN                    NaN  
59                           NaN                    NaN  
120                          NaN                    NaN  
143                          NaN                    NaN  
...                          ...                    ...  
19727                        NaN                    NaN  
19821                        NaN                    NaN  
19824                        NaN                    NaN  
19839                        NaN                    NaN  
19849                        NaN                    NaN  

[526 rows x 6 columns]

In [12]:
# unique data in host_is_superhost
df_listings.host_is_superhost.unique()

array(['f', 't', nan], dtype=object)

In [13]:
# unique data in host_total_listings_count
df_listings.host_total_listings_count.unique()

array([6.000e+00, 1.000e+00, 4.000e+00, 2.000e+00, 0.000e+00, 3.000e+00,
             nan, 9.000e+00, 5.000e+00, 2.000e+01, 8.000e+00, 7.000e+00,
       1.000e+01, 1.500e+01, 1.300e+01, 5.500e+01, 1.700e+01, 1.100e+01,
       2.500e+01, 1.800e+01, 1.200e+01, 1.900e+01, 1.400e+01, 4.600e+01,
       1.600e+01, 3.200e+01, 8.100e+01, 2.090e+02, 1.111e+03, 7.090e+02,
       2.200e+01, 3.800e+01, 2.350e+02, 1.510e+02, 3.030e+02, 4.400e+01,
       2.770e+02, 1.730e+02, 1.439e+03])

In [14]:
# unique data in host_identity_verified
df_listings.host_identity_verified.unique()

array(['t', 'f', nan], dtype=object)

In [15]:
# the data types of the elements in the columns
k = df_listings[df_listings.columns[df_listings.notnull().sum() == 10401]].columns
i = 0
for elem in df_listings.loc[1, df_listings.columns[df_listings.notnull().sum() == 10401]]:
    print(k[i] + "\t\t" + str(type(elem)))
    i = i + 1

host_since		<class 'str'>
host_is_superhost		<class 'str'>
host_thumbnail_url		<class 'str'>
host_picture_url		<class 'str'>
host_total_listings_count		<class 'numpy.float64'>
host_identity_verified		<class 'str'>


In the above columns, we take a sneak peak on what  the data and the data format looks like 

<a id='intro'></a>
### First Part Cleaning
[Link to summary](#summary)
#### From the above info cells we can see the types of the data and how the data looks like. In the following cells we will do the following changes: 

#### host_since
Data in column <i><b>host_since</b></i> will be replaced by the difference in days between the 12-03-2021 (the date this data was extracted) and the actual date in the column. The <b>NaN values</b> will be replaced by a random date say "2021-02-17", followed by the computing of the diffrence as above,  if the host has at least one review. The logic is that these accounts are fairly new and did not pass yet the verification  yet, since they have at least one review it means that they have at least one booking. Therefore they are real. The ones that do not have at least one review will be dropped<br /> 

#### host_is_superhost and host_identity_verified
<p>
The data in <i><b>host_is_superhost</b></i> and <i><b>host_identity_verified</b></i> will be changed to 1 for "t" and 0 for "f". The <b>NaN values</b> will be replaced by "f" meaning in this case 0 (zero). We assume for Nan values that it does not exist and will be therefore replaced by 0 (zero) <br /> 
</p>

[Link to cell](#step1)

#### host_thumbnail_url and host_picture_url
<p>
The data in <i><b>host_thumbnail_url</b></i> and <i><b>host_picture_url</b></i> will be changed to 1 when it exists and 0 for when it does not exist. The <b>NaN values</b> will be replaced by  0 (zero). We assume for Nan values that it does not exist 
</p>

[Link to cell](#step2)

#### host_total_listings_count
<p>
The data in <i><b>host_total_listings_count</b></i>  will not be changed. The <b>NaN values</b> will be replaced by  1 (one). We assume that since this listing exists it, this measure cannot be less than 1. 
</p>

[Link to cell](#step3)

### host_since 


In [16]:
# looking at the reviews per month
mask = (df_listings.host_since.isnull()) & (df_listings.number_of_reviews_ltm > 0)
df_listings.loc[mask, :]

id host_since host_about host_response_time host_response_rate  \
19        30295        NaN        NaN                NaN                NaN   
53        83614        NaN        NaN                NaN                NaN   
59        89102        NaN        NaN                NaN                NaN   
172      243238        NaN        NaN                NaN                NaN   
177      245991        NaN        NaN                NaN                NaN   
...         ...        ...        ...                ...                ...   
19078  46036795        NaN        NaN                NaN                NaN   
19209  46468883        NaN        NaN                NaN                NaN   
19402  47043089        NaN        NaN                NaN                NaN   
19568  47779522        NaN        NaN                NaN                NaN   
19607  47907303        NaN        NaN                NaN                NaN   

      host_acceptance_rate host_is_superhost host_thumbnail_url  \
19                     NaN               NaN                NaN   
53                     NaN               NaN                NaN   
59                     NaN               NaN                NaN   
172                    NaN               NaN                NaN   
177                    NaN               NaN                NaN   
...                    ...               ...                ...   
19078                  NaN               NaN                NaN   
19209                  NaN               NaN                NaN   
19402                  NaN               NaN                NaN   
19568                  NaN               NaN                NaN   
19607                  NaN               NaN                NaN   

      host_picture_url host_neighbourhood  ...  review_scores_rating  \
19                 NaN                NaN  ...                  95.0   
53                 NaN                NaN  ...                  96.0   
59                 NaN                NaN  ...                  97.0   
172                NaN                NaN  ...                  89.0   
177                NaN                NaN  ...                  95.0   
...                ...                ...  ...                   ...   
19078              NaN                NaN  ...                 100.0   
19209              NaN                NaN  ...                  70.0   
19402              NaN                NaN  ...                  80.0   
19568              NaN                NaN  ...                 100.0   
19607              NaN                NaN  ...                 100.0   

      review_scores_accuracy review_scores_cleanliness review_scores_checkin  \
19                      10.0                      10.0                  10.0   
53                      10.0                      10.0                  10.0   
59                      10.0                       9.0                  10.0   
172                     10.0                      10.0                  10.0   
177                     10.0                      10.0                  10.0   
...                      ...                       ...                   ...   
19078                   10.0                      10.0                  10.0   
19209                   10.0                       9.0                  10.0   
19402                   10.0                       8.0                   8.0   
19568                    8.0                      10.0                  10.0   
19607                   10.0                      10.0                  10.0   

      review_scores_communication  review_scores_location  \
19                           10.0                    10.0   
53                           10.0                    10.0   
59                           10.0                    10.0   
172                          10.0                     9.0   
177                           9.0                    10.0   
...                           ...                     ...   
19078 

In [17]:
# droping the rows with 0 reviews
mask = (df_listings.host_since.isnull()) & (df_listings.number_of_reviews_ltm == 0)
df_listings = df_listings.drop(df_listings.loc[mask, :].index)

In [18]:
# assigning the random date of '2021-02-17'
mask = (df_listings.host_since.isnull())
df_listings.loc[mask, 'host_since'] = '2021-02-17'

In [19]:
# checking if there are remaining NaN values 
df_listings[df_listings.host_since.isnull()]

Empty DataFrame
Columns: [id, host_since, host_about, host_response_time, host_response_rate, host_acceptance_rate, host_is_superhost, host_thumbnail_url, host_picture_url, host_neighbourhood, host_total_listings_count, host_identity_verified, neighbourhood, neighbourhood_cleansed, room_type, accommodates, bathrooms, bedrooms, beds, amenities, price, minimum_nights, maximum_nights, number_of_reviews, number_of_reviews_ltm, review_scores_rating, review_scores_accuracy, review_scores_cleanliness, review_scores_checkin, review_scores_communication, review_scores_location, review_scores_value, license, instant_bookable, calculated_host_listings_count_entire_homes]
Index: []

[0 rows x 35 columns]

In [20]:
# changing the column with the difference 
def change_date(date):
    days = datetime.strptime("2021-02-17", '%Y-%m-%d') - datetime.strptime(date, '%Y-%m-%d') 
    return days.days

df_listings.host_since = df_listings.host_since.apply(change_date)
df_listings.head()

id  host_since                                         host_about  \
1   2015        4566                       Isn’t sharing economy great?   
2   3176        4504  We love to travel ourselves a lot and prefer t...   
4   6883        4304  Hello and thanks for visitng my page. My name ...   
6   9991        4194  Born in Munich - lives in and loves Berlin. Wo...   
7  14325        4109           From Berlin, Germany. We love to travel!   

   host_response_time host_response_rate host_acceptance_rate  \
1      within an hour                97%                  95%   
2        within a day                86%                 100%   
4      within an hour               100%                 100%   
6  within a few hours               100%                  NaN   
7        within a day               100%                  67%   

  host_is_superhost                                 host_thumbnail_url  \
1                 f  https://a0.muscache.com/im/pictures/user/21428...   
2                 f  https://a0.muscache.com/im/users/3718/profile_...   
4                 t  https://a0.muscache.com/im/pictures/user/5df24...   
6                 f  https://a0.muscache.com/im/users/33852/profile...   
7                 t  https://a0.muscache.com/im/users/55531/profile...   

                                    host_picture_url host_neighbourhood  ...  \
1  https://a0.muscache.com/im/pictures/user/21428...              Mitte  ...   
2  https://a0.muscache.com/im/users/3718/profile_...    Prenzlauer Berg  ...   
4  https://a0.muscache.com/im/pictures/user/5df24...     Friedrichshain  ...   
6  https://a0.muscache.com/im/users/33852/profile...    Prenzlauer Berg  ...   
7  https://a0.muscache.com/im/users/55531/profile...    Prenzlauer Berg  ...   

   review_scores_rating review_scores_accuracy review_scores_cleanliness  \
1                  93.0                   10.0                       9.0   
2                  93.0                    9.0                       9.0   
4                  99.0                   10.0                      10.0   
6                 100.0                   10.0                      10.0   
7                  93.0                   10.0                      10.0   

  review_scores_checkin review_scores_communication  review_scores_location  \
1                  10.0                        10.0                    10.0   
2                   9.0                         9.0                    10.0   
4                  10.0                        10.0                    10.0   
6                  10.0                        10.0                    10.0   
7                   9.0                        10.0                     9.0   

   review_scores_value            license  instant_bookable  \
1                  9.0                NaN                 f   
2                  9.0                NaN                 f   
4                 10.0  02/Z/RA/008250-18                 f   
6                 10.0  03/Z/RA/003410-18                 f   
7                  9.0                NaN                 f   

  calculated_host_listings_count_entire_homes  
1                                           6  
2                                           1  
4                                           1  
6                                           1  
7                                           3  

[5 rows x 35 columns]

<a id = "step1"></a>
###  host_is_superhost and host_identity_verified 

[Link to intro](#intro)

In [21]:
# replacing nan values with 'f'
df_listings.host_is_superhost.fillna('f', inplace=True)
df_listings.host_identity_verified.fillna('f', inplace=True)

In [22]:
# replacing true and false values 
df_listings.host_is_superhost.replace('t', 1, inplace=True)
df_listings.host_is_superhost.replace('f', 0, inplace=True)
df_listings.host_identity_verified.replace('t', 1, inplace=True)
df_listings.host_identity_verified.replace('f', 0, inplace=True)

In [23]:
# checking the unique values
df_listings.host_is_superhost.unique()

array([0, 1])

In [24]:
# checking the unique values
df_listings.host_identity_verified.unique()

array([1, 0])

In [25]:
df_listings.head()

id  host_since                                         host_about  \
1   2015        4566                       Isn’t sharing economy great?   
2   3176        4504  We love to travel ourselves a lot and prefer t...   
4   6883        4304  Hello and thanks for visitng my page. My name ...   
6   9991        4194  Born in Munich - lives in and loves Berlin. Wo...   
7  14325        4109           From Berlin, Germany. We love to travel!   

   host_response_time host_response_rate host_acceptance_rate  \
1      within an hour                97%                  95%   
2        within a day                86%                 100%   
4      within an hour               100%                 100%   
6  within a few hours               100%                  NaN   
7        within a day               100%                  67%   

   host_is_superhost                                 host_thumbnail_url  \
1                  0  https://a0.muscache.com/im/pictures/user/21428...   
2                  0  https://a0.muscache.com/im/users/3718/profile_...   
4                  1  https://a0.muscache.com/im/pictures/user/5df24...   
6                  0  https://a0.muscache.com/im/users/33852/profile...   
7                  1  https://a0.muscache.com/im/users/55531/profile...   

                                    host_picture_url host_neighbourhood  ...  \
1  https://a0.muscache.com/im/pictures/user/21428...              Mitte  ...   
2  https://a0.muscache.com/im/users/3718/profile_...    Prenzlauer Berg  ...   
4  https://a0.muscache.com/im/pictures/user/5df24...     Friedrichshain  ...   
6  https://a0.muscache.com/im/users/33852/profile...    Prenzlauer Berg  ...   
7  https://a0.muscache.com/im/users/55531/profile...    Prenzlauer Berg  ...   

   review_scores_rating  review_scores_accuracy review_scores_cleanliness  \
1                  93.0                    10.0                       9.0   
2                  93.0                     9.0                       9.0   
4                  99.0                    10.0                      10.0   
6                 100.0                    10.0                      10.0   
7                  93.0                    10.0                      10.0   

  review_scores_checkin review_scores_communication  review_scores_location  \
1                  10.0                        10.0                    10.0   
2                   9.0                         9.0                    10.0   
4                  10.0                        10.0                    10.0   
6                  10.0                        10.0                    10.0   
7                   9.0                        10.0                     9.0   

   review_scores_value            license  instant_bookable  \
1                  9.0                NaN                 f   
2                  9.0                NaN                 f   
4                 10.0  02/Z/RA/008250-18                 f   
6                 10.0  03/Z/RA/003410-18                 f   
7                  9.0                NaN                 f   

  calculated_host_listings_count_entire_homes  
1                                           6  
2                                           1  
4                                           1  
6                                           1  
7                                           3  

[5 rows x 35 columns]

<a id = "step2"></a>
###  host_thumbnail_url and host_picture_url

[Link to intro](#intro)

In [26]:
#replacing the not nan with 1 
df_listings.loc[df_listings[df_listings.host_thumbnail_url.notna()].index, 'host_thumbnail_url'] = 1
df_listings.loc[df_listings[df_listings.host_picture_url.notna()].index, 'host_picture_url'] = 1

In [27]:
# replacing nan with 0 
df_listings.host_thumbnail_url.fillna(0, inplace=True)
df_listings.host_picture_url.fillna(0, inplace=True)



<a id = "step3"></a>
### host_total_listings_count
[Link to intro](#intro)

In [28]:
# changing nan values to 1 
df_listings.host_total_listings_count.fillna(0, inplace=True)

In [29]:
# view of the data after the changes
df_listings.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 10611 entries, 1 to 19856
Data columns (total 35 columns):
 #   Column                                       Non-Null Count  Dtype  
---  ------                                       --------------  -----  
 0   id                                           10611 non-null  int64  
 1   host_since                                   10611 non-null  int64  
 2   host_about                                   5536 non-null   object 
 3   host_response_time                           5061 non-null   object 
 4   host_response_rate                           5061 non-null   object 
 5   host_acceptance_rate                         6024 non-null   object 
 6   host_is_superhost                            10611 non-null  int64  
 7   host_thumbnail_url                           10611 non-null  int64  
 8   host_picture_url                             10611 non-null  int64  
 9   host_neighbourhood                           7336 non-null   object 
 10

<a id = "intro2"></a>
## Second Part Cleaning
[Link to summary](#summary)
### host_about
<p>
    In this column we are interested in the length of the about section and not the actual contents. The logic is that the lenght of the descirption, and not the actual contents might have an influence. The nan will be replaced by 0 
</p>

[Link to cell](#host_about)

### host_response_time

For the host response time we shall replace the nan values with "no response" and leave the other cells as they are. 

[Link to cell](#host_response_time)

### host_response_rate and host_acceptance_rate 

for this column we will extract the '97%' percent representation in the decimal form 0.97, the nan values will be replaced with 0

[Link to cell](#host_rate)

In [30]:
# a look at the host_about
df_listings.host_about.head()

1                         Isn’t sharing economy great?
2    We love to travel ourselves a lot and prefer t...
4    Hello and thanks for visitng my page. My name ...
6    Born in Munich - lives in and loves Berlin. Wo...
7             From Berlin, Germany. We love to travel!
Name: host_about, dtype: object

In [31]:
# unique values in host_response_time
df_listings.host_response_time.unique()


array(['within an hour', 'within a day', 'within a few hours', nan,
       'a few days or more'], dtype=object)

In [32]:
# a look at the host_response_rate
df_listings.host_response_rate.head()

1     97%
2     86%
4    100%
6    100%
7    100%
Name: host_response_rate, dtype: object

#### host_about
<a id = "host_about"></a>
[link to intro](#intro2)

In [33]:
df_listings.host_about

1                             Isn’t sharing economy great?
2        We love to travel ourselves a lot and prefer t...
4        Hello and thanks for visitng my page. My name ...
6        Born in Munich - lives in and loves Berlin. Wo...
7                 From Berlin, Germany. We love to travel!
                               ...                        
19848                                                  NaN
19852                            Traveler and Berlin fan. 
19853                                                  NaN
19854                                                  NaN
19856                                                  NaN
Name: host_about, Length: 10611, dtype: object

In [34]:
# creating a helper function
def about_transform(txt):
    if str(type(txt)) == "<class 'str'>":
        return len(txt)
    else:
        return 0

# applying the function, returning length or zero
df_listings.host_about = df_listings.host_about.apply(about_transform)

In [35]:
# checking the results 
df_listings.host_about.head()

1     28
2    102
4    299
6    228
7     40
Name: host_about, dtype: int64

#### host_response_time
<a id ="host_response_time"></a>
[link to intro](#intro2)

In [36]:

# filling nan values for  host_response_time 
df_listings.host_response_time.fillna('no response', inplace=True)

In [37]:
# checking results
df_listings.host_response_time

1            within an hour
2              within a day
4            within an hour
6        within a few hours
7              within a day
                ...        
19848        within an hour
19852    within a few hours
19853           no response
19854           no response
19856          within a day
Name: host_response_time, Length: 10611, dtype: object

#### host_response_rate and host_acceptance_rate
<a id = "host_rate"></a>
[link to intro](#intro2)

In [38]:
# creating the helper function
def convert_int(k):
    if str(type(k)) == "<class 'str'>":
        k = k.replace('%', '')
        return int(k)/100
    else:
        return 0
    
# applying the function to the host_response_rate
df_listings.host_response_rate = df_listings.host_response_rate.apply(convert_int)
# applying the function to the host_acceptance_rate
df_listings.host_acceptance_rate = df_listings.host_acceptance_rate.apply(convert_int)


In [39]:
# checking the results
df_listings.host_response_rate.head()

1    0.97
2    0.86
4    1.00
6    1.00
7    1.00
Name: host_response_rate, dtype: float64

In [40]:
# checking the results
df_listings.host_acceptance_rate.head()

1    0.95
2    1.00
4    1.00
6    0.00
7    0.67
Name: host_acceptance_rate, dtype: float64

In [41]:
# looking at the results after cleaning
df_listings.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 10611 entries, 1 to 19856
Data columns (total 35 columns):
 #   Column                                       Non-Null Count  Dtype  
---  ------                                       --------------  -----  
 0   id                                           10611 non-null  int64  
 1   host_since                                   10611 non-null  int64  
 2   host_about                                   10611 non-null  int64  
 3   host_response_time                           10611 non-null  object 
 4   host_response_rate                           10611 non-null  float64
 5   host_acceptance_rate                         10611 non-null  float64
 6   host_is_superhost                            10611 non-null  int64  
 7   host_thumbnail_url                           10611 non-null  int64  
 8   host_picture_url                             10611 non-null  int64  
 9   host_neighbourhood                           7336 non-null   object 
 10

<a id = "third" ></a>
## Third Part Cleaning 
[Link to summary](#summary)
### In the next part we will focus on columns "host_neighbourhood", "neighbourhood", "bathrooms",  "bedrooms" and "beds"

In [42]:
df_listings.iloc[:, [9, 12, 16, 17, 18]]

host_neighbourhood    neighbourhood  bathrooms  bedrooms  beds
1                  Mitte  Berlin, Germany        NaN       1.0   0.0
2        Prenzlauer Berg  Berlin, Germany        NaN       1.0   2.0
4         Friedrichshain  Berlin, Germany        NaN       1.0   1.0
6        Prenzlauer Berg  Berlin, Germany        NaN       4.0   7.0
7        Prenzlauer Berg              NaN        NaN       NaN   1.0
...                  ...              ...        ...       ...   ...
19848    Prenzlauer Berg              NaN        NaN       2.0   2.0
19852                NaN  Berlin, Germany        NaN       4.0   2.0
19853                NaN              NaN        NaN       1.0   1.0
19854                NaN              NaN        NaN       1.0   1.0
19856                NaN              NaN        NaN       1.0   NaN

[10611 rows x 5 columns]

####  host_neighbourhood and neighbourhood
looking at the data we can see the columns <i>host_neighbourhood</i>, <i>neighbourhood</i> and <i>neighbourhood_cleansed</i> practically convey the same data, we assume that the data that is cleansed by the Airbnb is the most reliable therefore the columns <i>host_neighbourhood</i> and  <i>neighbourhood</i> will be droped

[link to cell](#neighb)

#### bathrooms 
This column will likewise be deleted as all it's values are nan

[link to cell](#neighb)


#### bedrooms and beds 

For these columns, the assumption is that, if there is a nan value, there is no bedroom or bed respectively, therefore these nan values will be replace with 0

[link to cell](#beds)

<a id = "neighb"></a> 
#### host_neighbourhood, neighbourhood and  bathrooms

In [43]:
# droping the columns 
df_listings.drop(columns=['host_neighbourhood',  'neighbourhood', 'bathrooms'], inplace=True)

<a id = "beds"></a>
#### bedrooms and beds

In [44]:
df_listings.bedrooms.fillna(0, inplace=True)
df_listings.beds.fillna(0, inplace=True)

In [45]:
df_listings.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 10611 entries, 1 to 19856
Data columns (total 32 columns):
 #   Column                                       Non-Null Count  Dtype  
---  ------                                       --------------  -----  
 0   id                                           10611 non-null  int64  
 1   host_since                                   10611 non-null  int64  
 2   host_about                                   10611 non-null  int64  
 3   host_response_time                           10611 non-null  object 
 4   host_response_rate                           10611 non-null  float64
 5   host_acceptance_rate                         10611 non-null  float64
 6   host_is_superhost                            10611 non-null  int64  
 7   host_thumbnail_url                           10611 non-null  int64  
 8   host_picture_url                             10611 non-null  int64  
 9   host_total_listings_count                    10611 non-null  float64
 10

<a id="remain"></a>
### In this part we will be looking at the remaining columns that contain nan values 

In [46]:
# having a look at them 
df_listings.iloc[:, np.r_[21:29, 31]].head()

number_of_reviews_ltm  review_scores_rating  review_scores_accuracy  \
1                      4                  93.0                    10.0   
2                      2                  93.0                     9.0   
4                      2                  99.0                    10.0   
6                      0                 100.0                    10.0   
7                      1                  93.0                    10.0   

   review_scores_cleanliness  review_scores_checkin  \
1                        9.0                   10.0   
2                        9.0                    9.0   
4                       10.0                   10.0   
6                       10.0                   10.0   
7                       10.0                    9.0   

   review_scores_communication  review_scores_location  review_scores_value  \
1                         10.0                    10.0                  9.0   
2                          9.0                    10.0                  9.0   
4                         10.0                    10.0                 10.0   
6                         10.0                    10.0                 10.0   
7                         10.0                     9.0                  9.0   

   calculated_host_listings_count_entire_homes  
1                                            6  
2                                            1  
4                                            1  
6                                            1  
7                                            3

### Columns 21:29

For this columns, given that the data that belongs to these columns are numeric, the nan values will be assumend to be zero, therefore, the nan values will be filled with zero. 

### License 

For this column, the presence of the license should be marked with 1, while the abscense will be marked with 0. 

In [47]:
### changing nan values in columns 21 - 29
df_listings.iloc[:, np.r_[21:29, 31]] = df_listings.iloc[:, np.r_[21:29, 31]].fillna(0)

In [48]:
# creating a helper function
def license_transform(txt):
    if str(type(txt)) == "<class 'str'>":
        return 1 
    else:
        return 0

df_listings.license = df_listings.license.apply(license_transform)

In [49]:
df_listings.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 10611 entries, 1 to 19856
Data columns (total 32 columns):
 #   Column                                       Non-Null Count  Dtype  
---  ------                                       --------------  -----  
 0   id                                           10611 non-null  int64  
 1   host_since                                   10611 non-null  int64  
 2   host_about                                   10611 non-null  int64  
 3   host_response_time                           10611 non-null  object 
 4   host_response_rate                           10611 non-null  float64
 5   host_acceptance_rate                         10611 non-null  float64
 6   host_is_superhost                            10611 non-null  int64  
 7   host_thumbnail_url                           10611 non-null  int64  
 8   host_picture_url                             10611 non-null  int64  
 9   host_total_listings_count                    10611 non-null  float64
 10

Before moving on, we will transform the price from its current form '$42.00' as object to 42.00 as float 

In [50]:
# converting price to float
df_listings.price = df_listings.price.str.replace('$', '').str.replace(',', '').astype(float, errors = 'raise')

### The amenities cleaning 



In [51]:
# A quick view of the column 
df_listings.amenities

1        ["Smoke alarm", "Cooking basics", "Lockbox", "...
2        ["Dedicated workspace", "Hangers", "Kitchen", ...
4        ["Smoke alarm", "Cooking basics", "Dishes and ...
6        ["Dishes and silverware", "Kitchen", "Oven", "...
7        ["Dishes and silverware", "Host greets you", "...
                               ...                        
19848    ["Kitchen", "Hangers", "Smoke alarm", "Iron", ...
19852    ["Dedicated workspace", "Hangers", "Kitchen", ...
19853    ["Dedicated workspace", "Hangers", "Private en...
19854    ["Hot water kettle", "Dryer", "Hangers", "Hot ...
19856    ["Dedicated workspace", "Hangers", "Kitchen", ...
Name: amenities, Length: 10611, dtype: object

Since it looks like this column represents lists of user generated input, we will split the column by comma, then we will replce the characters such as [ ] " and space. In order to give the model a meaningfull representation we will create additional columns with the name of the amenities and a 0 or a 1 to mark the presence or absence of the amenity 

In [52]:
# creating a new df with splitted values 
df_split = df_listings.amenities.str.split(",", expand = True)

In [53]:
# function for creating dummies
def create_dummies_ammenities(df_model):
    df_object = df_model.select_dtypes(include='object').copy().columns
    for var in df_object:
        df_model = pd.concat([df_model.drop(var, axis = 1), pd.get_dummies(df_model[var],  drop_first=True)], axis = 1) 
    return df_model

In [54]:
# replacing the characters
for column in df_split.columns:
    df_split[column] =  df_split[column].str.replace('[', '').str.replace('"', '').str.replace(']', '').str.replace(' ', '')

In [55]:
# creating dummies out of df_split 
df_dummies = create_dummies_ammenities(df_split)

In [56]:
# droping ammenities from df_listings
df_listings.drop('amenities', axis = 'columns', inplace=True)

In [57]:
# creating a copy for later use 
df_prices = df_listings.copy()

In [58]:
# checking for any nulls 
df_dummies.isnull().any(axis = 1).sum()

0

In [59]:
# joining the dummies with the original df_listings 
df_listings = df_listings.join(df_dummies)

In [60]:
df_listings[df_listings.isnull().any(axis=1)]

Empty DataFrame
Columns: [id, host_since, host_about, host_response_time, host_response_rate, host_acceptance_rate, host_is_superhost, host_thumbnail_url, host_picture_url, host_total_listings_count, host_identity_verified, neighbourhood_cleansed, room_type, accommodates, bedrooms, beds, price, minimum_nights, maximum_nights, number_of_reviews, number_of_reviews_ltm, review_scores_rating, review_scores_accuracy, review_scores_cleanliness, review_scores_checkin, review_scores_communication, review_scores_location, review_scores_value, license, instant_bookable, calculated_host_listings_count_entire_homes, 32\HDTVwithstandardcable, Airconditioning, Bathroomessentials, Bedlinens, Boardgames, Bodysoap, CableTV, Carbonmonoxidealarm, Changingtable, Cleaningproducts, Cookingbasics, Dedicatedworkspace, Dishesandsilverware, Dryer, Elevator, Essentials, Ethernetconnection, Firstaidkit, Freeparkingonpremises, Freestreetparking, Freewasher\u2013Inbuilding, Gardenorbackyard, Gym, HDTV, Hairdryer, Hangers, Heating, Hostgreetsyou, Hotwater, Hotwaterkettle, Indoorfireplace, Iron, Keypad, Kitchen, Lockbox, Longtermstaysallowed, Luggagedropoffallowed, Microwave, Patioorbalcony, Petsallowed, Privateentrance, Refrigerator, Shampoo, Singlelevelhome, Smartlock, Smokealarm, Soundsystem, TV, Toaster, Washer, Washer\u2013\u00a0Inunit, Waterfront, Wifi, junorefrigerator, AEGinductionstove, Airconditioning, AmazonPrimeVideo, Babybath, Bakingsheet, Barbecueutensils, Bathroomessentials, Bedlinens, Bedroomcomforts, Bodysoap, BoseSoundlinkMinisoundsystemwithBluetoothandaux, Breakfast, Buildingstaff, CableTV, Carbonmonoxidealarm, ...]
Index: []

[0 rows x 4552 columns]

<a id = "modeling"></a>
## Modeling
<a href = "#summary"> Link to summary</a>

In this part we will try to answer the first question, first by trying to build a machine learning model and then by trying see the correlations with price <br />
### What are the factors that influence the price of the apartments ? 

#### Machine learning

In [61]:
# creating the dummy variables for the object 
## extracting the object columns 
# from now on we will work with dm_model
df_model = df_listings.copy()
df_object = df_model.select_dtypes(include='object').copy().columns

## creating the dummies 
def create_dummies(df_model):
    df_object = df_model.select_dtypes(include='object').copy().columns
    for var in df_object:
        df_model = pd.concat([df_model.drop(var, axis = 1), pd.get_dummies(df_model[var], prefix_sep="_", prefix=var, drop_first=True)], axis = 1) 
    return df_model

In [62]:
df_model = create_dummies(df_model)
df_model

id  host_since  host_about  host_response_rate  \
1          2015        4566          28                0.97   
2          3176        4504         102                0.86   
4          6883        4304         299                1.00   
6          9991        4194         228                1.00   
7         14325        4109          40                1.00   
...         ...         ...         ...                 ...   
19848  48579598         433           0                0.98   
19852  48595697        2971          25                1.00   
19853  48597066         -22           0                0.00   
19854  48599795         -22           0                0.00   
19856  48602039         771           0                1.00   

       host_acceptance_rate  host_is_superhost  host_thumbnail_url  \
1                      0.95                  0                   1   
2                      1.00                  0                   1   
4                      1.00                  1                   1   
6                      0.00                  0                   1   
7                      0.67                  1                   1   
...                     ...                ...                 ...   
19848                  0.40                  0                   1   
19852                  0.00                  0                   1   
19853                  0.00                  0                   1   
19854                  0.00                  0                   1   
19856                  1.00                  0                   1   

       host_picture_url  host_total_listings_count  host_identity_verified  \
1                     1                        6.0                       1   
2                     1                        1.0                       1   
4                     1                        1.0                       1   
6                     1                        1.0                       1   
7                     1                        4.0                       1   
...                 ...                        ...                     ...   
19848                 1                       22.0                       1   
19852                 1                        0.0                       1   
19853                 1                        0.0                       1   
19854                 1                        1.0                       1   
19856                 1                        0.0                       0   

       ...  neighbourhood_cleansed_West 4  neighbourhood_cleansed_West 5  \
1      ...                              0                              0   
2      ...                              0                              0   
4      ...                              0                              0   
6      ...                              0                              0   
7      ...                              0                              0   
...    ...                            ...                            ...   
19848  ...                              0                              0   
19852  ...                              0                              0   
19853  ...                              0                              0   
19854  ...                              0                              0   
19856  ...                              0                              0   

       neighbourhood_cleansed_Westend  \
1                                   0   
2                                   0   
4                                   0   
6                                   0   
7                                   0   
...                               ...   
19848                               0   
19852                               0   
19853                               0   
19854                               0   
19856                               0   

       neighbourhood_cleansed_Wiesbadener Straße  \
1                   

In [63]:
#Spliting into explanatory and response variables

y = df_model.price 
X = df_model.drop(columns=['price', 'id'])
def model_function(X, y):
    #Spliting into train and test
    X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.33, random_state=42)

    # Instantiate
    lm_model = LinearRegression(normalize=True)
    #Fit
    lm_model.fit(X_train, y_train)
    #Predict and score the model
    y_test_predics = lm_model.predict(X_test)
    
    return(lm_model, y_test, y_test_predics)

model, y_test, y_test_predics = model_function(X, y)
    ### prediction score
print("The prediction score is {}".format(r2_score(y_test, y_test_predics)))

The prediction score is -6.771935298994882e+27


The score is bigger than 1, a clear indication that there are too many features and too few observations
Next, we will try to see the corelation to price only

In [64]:
# creating a copy

df_model = df_listings.copy()

In [65]:
df_model.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 10611 entries, 1 to 19856
Columns: 4552 entries, id to Extrapillowsandblankets
dtypes: float64(13), int64(14), object(4), uint8(4521)
memory usage: 48.7+ MB


#### Corellation analysis

In [66]:
price_corelation = df_model.corr()['price']

In [67]:
price_corelation[price_corelation > 0.1]

accommodates        0.332795
bedrooms            0.297032
beds                0.262674
price               1.000000
Indoorfireplace     0.119097
Outdoorfurniture    0.130624
Hottub              0.149923
Indoorfireplace     0.109254
Buildingstaff       0.110839
Name: price, dtype: float64

Here we can see a very weak corelation for the accomodates, bedrooms and beds.

Now we move on to the second and third question: 

###  What are the neighbourhoods with the highest priced appartments ? 
###  What are the neighbourhoods with the lowest priced appartments ? 

but before that we need to extract the relevant data

In [68]:
# selecting the columns that are relevant for the questions
df_prices = df_prices[['price', 'number_of_reviews_ltm', 'neighbourhood_cleansed']]

In [69]:
df_prices.head()

price  number_of_reviews_ltm    neighbourhood_cleansed
1   59.0                      4           Brunnenstr. Süd
2   90.0                      2   Prenzlauer Berg Südwest
4   79.0                      2  Frankfurter Allee Süd FK
6  180.0                      0   Prenzlauer Berg Südwest
7   70.0                      1  Prenzlauer Berg Nordwest

In [70]:
# the view of the data with the quartiles
df_prices.describe()

price  number_of_reviews_ltm
count  10611.000000           10611.000000
mean      85.638865               2.740835
std       95.694979               7.121375
min        9.000000               0.000000
25%       50.000000               0.000000
50%       70.000000               0.000000
75%       98.000000               2.000000
max     4000.000000             222.000000

In order to have an accurate view of the prices of the appartments, I will look at the median of the apartments, also, in order to have a better view, I will eliminate the appartments that have no reviews. In my opinon this is a proxy that the appartments was listed but never actually rented. To be more specific why am I doing this, I want to eliminate the cases where there are aparmtents listed for let's say 1 million dollars per night. While this might be a possibility, in Berlin no one will rent such an apartment. 

In [71]:

df_prices = df_prices[df_prices['number_of_reviews_ltm' ] > 1]

In [72]:
# selecting only the columns that are of interest to us 
df_prices  = df_prices[['price', 'neighbourhood_cleansed']]

In [73]:
# visualizing quartiles
df_prices.describe()

price
count  3063.000000
mean     84.561867
std      54.965716
min      14.000000
25%      50.000000
50%      70.000000
75%      99.000000
max     650.000000

In [74]:
# selecting the apartments in the 1st quartile
df_prices_25 = df_prices[df_prices.price < 50]

In [75]:
# selecting the apartments in the 4th quartile 
df_price_75 = df_prices[df_prices.price > 98]

In [76]:
df_prices_25.head()

price neighbourhood_cleansed
9     49.0   südliche Luisenstadt
49    49.0          Schmargendorf
71    48.0        Schöneberg-Nord
89    47.0         Helmholtzplatz
104   43.0        Neue Kantstraße

In [77]:
# 1st quartile prices grouped by the mean of the neighbourhood
df_prices_25.groupby('neighbourhood_cleansed').mean().sort_values(by = 'price').head()

price
neighbourhood_cleansed                          
West 2                                      25.0
Plänterwald                                 25.5
Schmöckwitz/Karolinenhof/Rauchfangswerder   26.0
Lankwitz                                    27.0
Heerstraße Nord                             29.0

In [78]:
# 4th quartile prices grouped by the mean of the neighbourhood
df_price_75.groupby('neighbourhood_cleansed').mean().sort_values(by = 'price', ascending=False).head()

price
neighbourhood_cleansed          
Köpenick-Süd               650.0
Grünau                     300.0
Alt-Hohenschönhausen Nord  299.0
Baumschulenweg             297.0
Rummelsburger Bucht        252.0

Including all the neighbourhoods in this calculation is faulty, as there might be few apartments per neighbourhood, as count, that distort the data. In other words, I have compared neighbourhoods were there are 3 apartments with the ones that have 300. <br />

Given this, I will do the calculation for the top 20 apartments by count

In [79]:
# now we select only the top 20 neighbourhoods 
top_20_count = df_prices.groupby('neighbourhood_cleansed').count().sort_values(by = 'price', ascending=False).head(30).index
top_20_count = list(top_20_count)

In [80]:
# keeping only those in the top 20
df_prices = df_prices[df_prices.neighbourhood_cleansed.isin(top_20_count)]


In [81]:
# showing the neighbourhoods with the lowest average prices top 10 cheapest first 
df_prices[df_prices['price'] < df_prices.describe().loc['25%', :]['price']].groupby(by = 'neighbourhood_cleansed').mean().sort_values(by = 'price').head(10)

price
neighbourhood_cleansed             
Düsseldorfer Straße       37.600000
Karl-Marx-Allee-Nord      38.750000
Rixdorf                   39.142857
Moabit Ost                39.833333
Karl-Marx-Allee-Süd       39.888889
Helmholtzplatz            39.969697
Frankfurter Allee Süd FK  40.178571
Moabit West               40.384615
Prenzlauer Berg Süd       40.882353
südliche Luisenstadt      41.687500

In [82]:
# showing the neighbourhoods with the lowest average prices top 10 most expensive first 
df_prices[df_prices['price'] > df_prices.describe().loc['75%', :]['price']].groupby(by = 'neighbourhood_cleansed').mean().sort_values(by = 'price', ascending = False).head(10)

price
neighbourhood_cleansed              
Neuköllner Mitte/Zentrum  223.900000
Tempelhof                 215.666667
Moabit West               205.476190
Reuterstraße              187.333333
Moabit Ost                184.090909
nördliche Luisenstadt     183.333333
Prenzlauer Berg Nord      179.333333
Prenzlauer Berg Süd       168.117647
Karl-Marx-Allee-Süd       164.777778
Schöneberg-Süd            164.055556

<a id = "evaluation"></a>
## Evaluation
<a href = "#summary"> Link to summary</a>

<p>
    The main takeaways from this notebook is that in the Airbnb data there are too many features to determine a plausible model and that there is a weak corelation between the price and the number of persons an apartment accomodates. 
</p> 
<p>
    When it comes to the most expensive vs cheapest apartments on average data is also inconclusive. As locations determined by the available grouping from Airbnb appear both in the most expensive (avg) and cheapest (avg) top 10 lists. 
</p>

<p>
   Given the weak correlation between the prices and the numbers of persons that an appartment accomodates, we might conclude that there is a posibility that Airbnb Berlin renters usually come to berlin in groups, looking for appartments that accomodate more people at once.  
</p>